src/main/java/ru/l1ttleO/boyfriend/commands/Clear.java

```diff
@@ -20,43 +20,57 @@
 
 import java.util.List;
 import net.dv8tion.jda.api.Permission;
-import net.dv8tion.jda.api.entities.GuildChannel;
+import net.dv8tion.jda.api.entities.Member;
 import net.dv8tion.jda.api.entities.Message;
-import net.dv8tion.jda.api.entities.MessageChannel;
+import net.dv8tion.jda.api.entities.TextChannel;
 import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 import ru.l1ttleO.boyfriend.Actions;
+import ru.l1ttleO.boyfriend.I18n.BotLocale;
 import ru.l1ttleO.boyfriend.Utils;
-import ru.l1ttleO.boyfriend.exceptions.InvalidAuthorException;
+import ru.l1ttleO.boyfriend.commands.util.CommandReader;
+import ru.l1ttleO.boyfriend.commands.util.Sender.ConsoleSender;
+import ru.l1ttleO.boyfriend.commands.util.Sender.MessageSender;
 import ru.l1ttleO.boyfriend.exceptions.NoPermissionException;
 import ru.l1ttleO.boyfriend.exceptions.WrongUsageException;
+import ru.l1ttleO.boyfriend.settings.GuildSettings;
 
-public class Clear extends Command {
+import static ru.l1ttleO.boyfriend.I18n.tl;
+
+public class Clear extends Command implements IChatCommand, IConsoleCommand {
 
     public Clear() {
-        super("clear", "Удаляет указанное количество сообщений в канале", "clear <количество, не менее 1 и не больше 99>");
+        super("clear", Permission.MESSAGE_MANAGE);
+    }
+
+    @Override
+    public void run(final @NotNull MessageReceivedEvent event, final @NotNull CommandReader reader, final @NotNull MessageSender sender)
+        throws NoPermissionException, WrongUsageException {
```

Comment:
```suggestion
        throws WrongUsageException {
```
`The declared exception 'NoPermissionException' is never thrown` [intellij]

get an actual IDE I stg